init.lua

```diff
@@ -89,19 +89,51 @@ function torch.type(obj)
 end
 
 --[[ See if a given object is an instance of the provided torch class. ]]
+local function exactTypeMatch(obj_mt, typeSpecNameString)
+  return obj_mt.__typename == typeSpecNameString
+end
+
+local function partialTypeMatch(obj_mt, typeSpecNameString)
```

Comment:
After re-reading: correct me if I'm wrong - this is effectively just checking that obj_mt.__typename is a suffix of typeSpecNameString ? Then that can certainly be accomplished with a single comparison of obt_mt.__typename with typeSpecNameString:sub(...) ?